flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -66,8 +70,52 @@ public Fabric8FlinkKubeClient(Configuration flinkConfig, KubernetesClient client
 	}
 
 	@Override
-	public void createTaskManagerPod(TaskManagerPodParameter parameter) {
-		// todo
+	public void createJobManagerComponent(KubernetesJobManagerSpecification kubernetesJMSpec) {
+		final Deployment deployment = kubernetesJMSpec.getDeployment();
+		final List<HasMetadata> accompanyingResources = kubernetesJMSpec.getAccompanyingResources();
+
+		// create Deployment
+		LOG.debug("Start to create deployment with spec {}", deployment.getSpec().toString());
+		final Deployment createdDeployment = this.internalClient
```

Comment:
We create the deployment first, then the accompanyingResources(e.g. configmap, service) are created. Does it have chance that the deployment is created with failure? Because the it needs to mount an nonexistent `ConfigMap`.

Reply:
The deployment would never fail since the Kubernetes has strategies for such a case:
1. Generally, the kubelet component helps retry pulling the missing ConfigMap.
2. The controller helps re-deploy the Pod if strategy 1 fails.